### ISSUE TITLE
Token based authenticaion not documened

### ISSUE THREAD

**Issue by i-g-g-y** [marvinroger/async-mqtt-client#291]
Hello.

I can't seem to find the token authentication method?

Is this supported?

I am using Beebotte and this is what it says about authentication: https://beebotte.com/docs/mqtt

**Comment by bertmelis** [comment]
there's nothing special about this from an MQTT point of view. Just set the username to the token. I don't know if you need to set the password to an empty string `const char* passwd = "";` or copy your token as the documentation says.

**Comment by i-g-g-y** [comment]
You are right about that. I rolled the dice and put the token as user name as string: "token_user" and as password I omitted the parameter:

`
Async_Mqtt_Client.setCredentials("token_blabla123");
`

This worked for Bebotte.

Thanks!
